### PR TITLE
fix: Added coverage

### PR DESCRIPTION
**Pull request by tmayoff**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -93,6 +94,10 @@
               # Documentation
               doxygen
               graphviz
+
+              # Coverage
+              gcovr
+              # lcov
             ]
             ++ rawNativeBuildInputs;
```
Comment (Artturin) [inline flake.nix:103]: You don't have boost in buildInputs, the variables you're setting are used by build systems but not by nixpkgs wrappers

File: flake.nix [hunk 2]
```diff
@@ -33,7 +33,7 @@
         ];
 
         rawBuildInputs = with pkgs; [
-          boost.dev
```
Comment (Artturin) [inline flake.nix:36]: They're the same thing btw https://github.com/NixOS/nixpkgs/blob/2b2a8530a8c6461205e0a2c7b839d5c7502454c1/pkgs/stdenv/generic/make-derivation.nix#L312 `getDev`

File: flake.nix (hunk 1 above)
Comment (Artturin) [reply flake.nix:103]: Maybe you can find something in https://github.com/search?q=lang%3Anix+NOT+is%3Afork+gcov&type=code

File: flake.nix (hunk 2 above)
Comment (tmayoff) [reply flake.nix:36]: Thanks, I had added the .dev as a try anything fix for something, now reverted back to without .dev

File: flake.nix (hunk 1 above)
Comment (tmayoff) [reply flake.nix:103]: I'm trying gcovr instead of llvm-cov, llvm-cov had other issues I couldn't figure out

Comment (tmayoff) [reply flake.nix:103]: For reference the issues with llvm-cov

```
ninja: Entering directory `build'
[1/1] Generates coverage reports
FAILED: meson-internal__coverage 
/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/bin/meson --internal coverage /home/tyler/src/wren /home/tyler/src/wren/subprojects /home/tyler/src/wren/build /home/tyler/src/wren/build/meson-logs --llvm-cov llvm-cov --use-llvm-cov
LLVM (http://llvm.org/):
  LLVM version 17.0.6
  Optimized build.
.genhtml-wrapped: LCOV version 2.1-beta
Capturing coverage data from /home/tyler/src/wren/build
geninfo cmd: '/nix/store/klla20c44lfc0w5cs9nwa9v62x5wfhdd-lcov-2.1/bin/geninfo /home/tyler/src/wren/build --output-filename /home/tyler/src/wren/build/meson-logs/coverage.info.initial --gcov-tool /home/tyler/src/wren/build/meson-logs/llvm-cov.sh --initial'
Found LLVM gcov version 17.0.6, which emulates gcov version 4.2.0
Using intermediate gcov format
Recording 'internal' directories:
	/home/tyler/src/wren/build
Writing temporary data to /tmp/geninfo_datc2HG
Scanning /home/tyler/src/wren/build for .gcno files ...
.geninfo-wrapped: WARNING: (usage) branch filter enabled but branch coverage not enabled
	(use ".geninfo-wrapped --ignore-errors usage,usage ..." to suppress this warning)
Found 158 graph files in /home/tyler/src/wren/build
using: chunkSize: 1, nchunks:158, intervalLength:7
.geninfo-wrapped: WARNING: (gcov) GCOV did not produce any data for /home/tyler/src/wren/build/subprojects/flecs-4.0.2/libflecs.so.p/src_addons_journal.c.gcno
	(use ".geninfo-wrapped --ignore-errors gcov,gcov ..." to suppress this warning)
.geninfo-wrapped: WARNING: (gcov) GCOV did not produce any data for /home/tyler/src/wren/build/wren_vk/libwren_vk.so.p/src_vulkan.cpp.gcno
.geninfo-wrapped: WARNING: (gcov) GCOV did not produce any data for /home/tyler/src/wren/build/wren/libwren.so.p/src_event.cpp.gcno
Finished processing 158 GCNO files
Apply filtering..
Message summary:
  1 error message:
    source: 1
  4 warning messages:
    gcov: 3
    usage: 1
.geninfo-wrapped: ERROR: (source) unable to open /home/tyler/src/wren/build/include/boost/test/tools/floating_point_comparison.hpp: No such file or directory
	(use ".geninfo-wrapped --ignore-errors source ..." to bypass this error)
Traceback (most recent call last):
  File "/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/bin/meson", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/lib/python3.12/site-packages/mesonbuild/mesonmain.py", line 291, in main
    return run(sys.argv[1:], launcher)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/lib/python3.12/site-packages/mesonbuild/mesonmain.py", line 279, in run
    return run_script_command(args[1], args[2:])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/lib/python3.12/site-packages/mesonbuild/mesonmain.py", line 220, in run_script_command
    return module.run(script_args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/lib/python3.12/site-packages/mesonbuild/scripts/coverage.py", line 208, in run
    return coverage(options.outputs, options.source_root,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/rrpijgwiw3y1qfvbyzi18qiap02fkp3r-meson-1.5.2/lib/python3.12/site-packages/mesonbuild/scripts/coverage.py", line 113, in coverage
    subprocess.check_call([lcov_exe,
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['lcov', '--directory', '/home/tyler/src/wren/build', '--capture', '--initial', '--output-file', '/home/tyler/src/wren/build/meson-logs/coverage.info.initial', '--gcov-tool', '/home/tyler/src/wren/build/meson-logs/llvm-cov.sh']' returned non-zero exit status 1.
ninja: build stopped: subcommand failed.

```

**Pull request merged**